packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -196,7 +202,12 @@ export abstract class SsoAccessTokenProvider {
             // We don't want to attribute the wrong source.
             SsoAccessTokenProvider.authSource = 'unknown'
 
-            return func()
+            const result = await func()
+
+            // The browser login process has succeeded, the reason is now stale
+            await ReAuthReasonState.instance.clearReason(this.profile)
```

Comment:
is it possible this doesn't get called? then the "wrong" state may be used in the future